instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/ContainerResource.java

```diff
@@ -57,33 +53,28 @@ public static Resource get() {
    * @return containerId
    */
   @IgnoreJRERequirement
-  @Nullable
-  private static String extractContainerId(Path cgroupFilePath) {
+  private static Optional<String> extractContainerId(Path cgroupFilePath) {
     if (!Files.exists(cgroupFilePath) || !Files.isReadable(cgroupFilePath)) {
-      return null;
+      return Optional.empty();
     }
     try (Stream<String> lines = Files.lines(cgroupFilePath)) {
-      Optional<String> value =
-          lines
-              .filter(line -> !line.isEmpty())
-              .map(ContainerResource::getIdFromLine)
-              .filter(Objects::nonNull)
-              .findFirst();
-      if (value.isPresent()) {
-        return value.get();
-      }
+      return lines
+          .filter(line -> !line.isEmpty())
+          .map(ContainerResource::getIdFromLine)
+          .filter(Optional::isPresent)
+          .findFirst()
+          .orElse(Optional.empty());
```

Comment:
Wouldn't
```suggestion
          .flatMap(ContainerResource::getIdFromLine);
```
work here?

Reply:
No, unfortunately `flatMap()` won't work here because `ContainerResource::getIdFromLine` returns `Optional<String>` and we end up with `Stream<Optional<String>>`. It would work as the last step though, to "pop" the `Optional<Optional<String>>`:
```suggestion
          .map(ContainerResource::getIdFromLine)
          .filter(Optional::isPresent)
          .findFirst()
          .flatMap(identity());
```
Is the last line clearer this way?